omod/src/main/java/org/openmrs/module/appointmentscheduling/rest/search/AppointmentSearchHandler.java

```diff
@@ -0,0 +1,76 @@
+package org.openmrs.module.appointmentscheduling.rest.search;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Location;
+import org.openmrs.Provider;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointmentscheduling.Appointment;
+import org.openmrs.module.appointmentscheduling.AppointmentType;
+import org.openmrs.module.appointmentscheduling.api.AppointmentService;
+import org.openmrs.module.appointmentscheduling.rest.controller.AppointmentRestController;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Date;
+
+@Component
+public class AppointmentSearchHandler  implements SearchHandler {
```

Comment:
Let's rename this "EarlyAndLateAppointmentSearchHandler" to reflect that it's specifically for handling those two searches, thanks!